ecosystem/nft-metadata-crawler/nft-metadata-crawler-parser/src/parser_entry.rs

```diff
@@ -168,21 +174,67 @@ impl ParserEntry {
     }
 
     /// Calls and handles error for writing JSON to GCS
-    async fn handle_write_json_to_gcs(&mut self, _json: Option<Value>) -> Option<String> {
-        todo!();
+    async fn handle_write_json_to_gcs(&mut self, json: Option<Value>) -> Option<String> {
+        if let Some(json) = json {
+            match write_json_to_gcs(
+                self.token.clone(),
+                self.bucket.clone(),
+                self.entry.token_data_id.clone(),
+                json,
+            )
+            .await
+            {
+                Ok(filename) => {
+                    self.log_info("Successfully wrote JSON to GCS");
+                    Some(filename)
+                },
+                Err(e) => {
+                    self.log_error(&e.to_string());
+                    None
+                },
+            }
+        } else {
+            self.log_error("No JSON to write to GCS");
+            None
+        }
     }
 
     /// Calls and handles error for writing image to GCS
     async fn handle_write_image_to_gcs(
         &mut self,
-        _image: Option<(Vec<u8>, ImageFormat)>,
+        image: Option<(Vec<u8>, ImageFormat)>,
     ) -> Option<String> {
-        todo!();
+        if let Some((image, format)) = image {
+            match write_image_to_gcs(
+                self.token.clone(),
+                format,
+                self.bucket.clone(),
+                self.entry.token_data_id.clone(),
+                image,
+            )
+            .await
+            {
+                Ok(filename) => {
+                    self.log_info("Successfully wrote image to GCS");
+                    Some(filename)
+                },
+                Err(e) => {
+                    self.log_error(&e.to_string());
+                    None
+                },
+            }
+        } else {
+            self.log_error("No image to write to GCS");
+            None
+        }
     }
 
     /// Calls and handles error for upserting to Postgres
     async fn commit_to_postgres(&mut self) {
-        todo!();
+        match upsert_uris(&mut self.conn, self.model.clone()) {
+            Ok(_) => self.log_info("Successfully committed to Postgres"),
+            Err(e) => self.log_error(&format!("Failed to commit to Postgres: {}", e)),
+        };
     }
```

Comment:
You'll find that this error logging won't help you that much without the query itself. Hence I was telling you to use something like this 
https://github.com/bowenyang007/aptos-core/blob/main/crates/indexer/src/database.rs#L64-L89